test/src/test/java/hudson/model/UpdateCenterErrorsTest.java

```diff
@@ -0,0 +1,185 @@
+package hudson.model;
+
+import com.gargoylesoftware.htmlunit.Page;
+import com.gargoylesoftware.htmlunit.html.DomNodeList;
+import com.gargoylesoftware.htmlunit.html.HtmlButton;
+import com.gargoylesoftware.htmlunit.html.HtmlElement;
+import com.gargoylesoftware.htmlunit.html.HtmlElementUtil;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import hudson.Messages;
+import hudson.util.HttpResponses;
+import jenkins.model.Jenkins;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.xml.sax.SAXException;
+
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertEquals;
+
+public class UpdateCenterErrorsTest {
+    private static final Logger LOGGER = Logger.getLogger(Jenkins.class.getName());
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    /**
+     * Check if the page contains the right message after checking an update site with an url that returns a 502 error code.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteReturn502Test() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSite502/getJson", "Server returned HTTP response code: 502 for URL", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site with an url that returns a wrong json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteWrongJsonTest() throws Exception {
+        checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteWrongJson/getJson", "Unquotted string 'wrongjson'", false);
+    }
+
+    /**
+     * Check if the page contains the right message after checking an update site that returns a well defined json.
+     * @throws Exception If there are errors managing the web elements.
+     */
+    @Test
+    public void updateSiteRightJsonTest() throws Exception {
+        //Avoid CertPathValidatorException: Algorithm constraints check failed on signature algorithm: MD5withRSA
+        DownloadService.signatureCheck = false;
```

Comment:
Perhaps worth using try{...} finally{...} to put back the initial value after that as it could have side effect?

(no suggestion as it modifies the whole code in the method ^^)

Reply:
{{@Before}} / {{@After}} are global for the tests but those configurations seem to be related to this specific one.

To be more precise about my proposal:
````java
boolean currentValue = DownloadService.signatureCheck;
try{ 
   DownloadService.signatureCheck = false;
   checkUpdateSite(Jenkins.getInstance().getRootUrl() + "updateSiteRightJson/update-center.json", "", true );
}
finally{
   DownloadService.signatureCheck = currentValue;
}
```

With that approach we do not let side effects in place after the run. (not sure it has any)